garmin_gpi.cc

```diff
@@ -820,48 +826,45 @@ wdata_free(writer_data_t* data)
     wdata_free(data->buttom_right);
   }
 
-  xfree(data);
+  delete data;
 }
 
 
 static void
 wdata_add_wpt(writer_data_t* data, Waypoint* wpt)
 {
-  data->ct++;
-  ENQUEUE_TAIL(&data->Q, &wpt->Q);
+  data->waypt_list.append(wpt);
   waypt_add_to_bounds(&data->bds, wpt);
 }
 
 
 static void
 wdata_check(writer_data_t* data)
 {
-  queue* elem, *tmp;
   double center_lon;
 
-  if ((data->ct <= WAYPOINTS_PER_BLOCK) ||
+  if ((data->waypt_list.size() <= WAYPOINTS_PER_BLOCK) ||
       /* avoid endless loop for points (more than WAYPOINTS_PER_BLOCK)
          at same coordinates */
       ((data->bds.min_lat >= data->bds.max_lat) && (data->bds.min_lon >= data->bds.max_lon))) {
-    if (data->ct > 1) {
-      sortqueue(&data->Q, compare_wpt_cb);
+    if (data->waypt_list.size() > 1) {
+      std::stable_sort(data->waypt_list.begin(), data->waypt_list.end(), compare_wpt_cb);
```

Comment:
We can use std:: on a QList. Huh. I wouldn't have thought to try that. That's cool!

Reply:
Yes, and it is official: https://doc.qt.io/qt-5/containers.html#introduction

> For traversing the items stored in a container, you can use one of two types of iterators: Java-style iterators and STL-style iterators. The Java-style iterators are easier to use and provide high-level functionality, whereas the STL-style iterators are slightly more efficient and can be used together with Qt's and STL's generic algorithms.